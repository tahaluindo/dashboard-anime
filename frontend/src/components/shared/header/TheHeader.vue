<template>
  <header
    ref="header"
    class="
      fixed
      z-50
      flex
      justify-between
      w-full
      h-10
      px-3
      py-2
      lg:h-16
      lg:px-5
      lg:py-3
      bg-primary
    "
  >
    <slot name="logo" />
    <the-header-nav-mobile v-if="mobile">
      <slot name="navMobile" />
    </the-header-nav-mobile>
    <the-header-nav v-else>
      <slot name="nav" />
    </the-header-nav>
  </header>
</template>

<script lang="ts">
import { defineComponent, ref } from "vue";
import TheHeaderNav from "./TheHeaderNav.vue";
import TheHeaderNavMobile from "./TheHeaderNavMobile.vue";

export default defineComponent({
  components: {
    TheHeaderNav,
    TheHeaderNavMobile,
  },
  setup() {
    const header = ref(null as unknown as HTMLElement);
    const mobile = ref(true);

    return {
      mobile,
      header,
      close,
    };
  },
});
</script>
