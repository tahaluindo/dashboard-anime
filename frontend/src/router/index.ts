import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import AddUser from "../views/AddUser.vue";
import AllUsers from "../views/AllUsers.vue";
import Home from "../views/Home.vue";
import User from "../views/User.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/usuarios",
    name: "AllUsers",
    component: AllUsers,
  },
  {
    path: "/usuario/:id",
    component: User,
    name: "User",
    props: true,
  },
  {
    path: "/adicionar-usuario",
    name: "AddUser",
    component: AddUser,
  },
];

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes,
  scrollBehavior() {
    return window.scrollTo({ top: 0, behavior: "smooth" });
  },
});

export default router;
