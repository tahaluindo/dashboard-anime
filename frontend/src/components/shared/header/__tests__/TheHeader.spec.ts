import { mount } from "@vue/test-utils";
import router from "../../../../router/__mocks__/router.mock";
import {
  mockWindowIntersectionObserver,
  mockWindowVisualViewport,
} from "../../../../__mocks__/window.mock";
import {
  HeaderObserver,
  HeaderObserverKey,
} from "../controllers/HeaderObserver";
import TheHeader from "../TheHeader.vue";

describe("TheHeader suit test", () => {
  let headerObserver: HeaderObserver;

  beforeEach(() => {
    mockWindowIntersectionObserver();
    mockWindowVisualViewport();

    headerObserver = new HeaderObserver();
  });

  it("should render the component with this default state", async () => {
    router.push("/contato");
    await router.isReady();

    const wrapper = mount(TheHeader, {
      global: {
        stubs: {
          TheHeaderNav: true,
          TheHeaderNavMobile: true,
          TheHeaderLogo: true,
        },
        provide: {
          [HeaderObserverKey as symbol]: headerObserver,
        },
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
