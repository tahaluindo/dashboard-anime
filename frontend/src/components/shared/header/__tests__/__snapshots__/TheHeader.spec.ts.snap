// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader suit test should render the component with this default state 1`] = `
"<header class=\\"
      fixed
      z-50
      flex
      justify-between
      w-full
      h-10
      px-3
      py-2
      transition
      lg:h-16
      duration-450
      lg:px-5
      lg:py-3
    \\">
  <the-header-nav-mobile-stub></the-header-nav-mobile-stub>
</header>"
`;
